smokey/src/main/java/frc/robot/subsystems/NavxSubsystem.java

```diff
@@ -0,0 +1,67 @@
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Charged Up - 2023
+// File: NavxSubsystem.java
+// Intent: Forms a stub for the prelminary named subsystem above.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.subsystems;
+
+import static frc.robot.Constants.*;
+import com.kauailabs.navx.frc.AHRS;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class NavxSubsystem extends SubsystemBase {
+
+  // The important thing about how you configure your gyroscope is that rotating the robot counter-clockwise should
+  // cause the angle reading to increase until it wraps back over to zero.
+  private final AHRS swerveNavx = new AHRS(SPI.Port.kMXP, (byte) 200); // NavX connected over MXP
+
+  /** Creates a new NavX. */
+  public NavxSubsystem() {}
+
+  /**
+   * Sets the gyroscope angle to zero. This can be used to set the direction the robot is currently facing to the
+   * 'forwards' direction.
+   */
+  public void zeroGyroscope() {
+        swerveNavx.zeroYaw();
+  }
+
+  /**
+   * Obtains the current gyroscope's rotation about the Z axis when looking down at the robot where positive is measured in the 
+   * counter-clockwise direction.
+   * 
+   * Notes: 
+   * According to https://pdocs.kauailabs.com/navx-mxp/wp-content/uploads/2020/09/navx2-mxp_robotics_navigation_sensor_user_guide-8.pdf
+   * NavX .getYaw() is -180 to 180 where positive is clockwise looking at the sensor from above
+   * According to https://docs.wpilib.org/en/stable/docs/software/advanced-controls/geometry/coordinate-systems.html#field-coordinate-system
+   * positive rotations are measured in the counter-clockwise looking at the robot from above
+   * these two pieces of information imply the yaw should be inverted (or multiplied by -1.0)
+   * @return A Rotation2d that describes the current orentation of the robot.
+   */
+  public Rotation2d getGyroscopeRotation() {
+    if (swerveNavx.isMagnetometerCalibrated()) {
+      // We will only get valid fused headings if the magnetometer is calibrated
+      return Rotation2d.fromDegrees(swerveNavx.getFusedHeading());
```

Comment:
I believe line 51 is the one that always gets called and line 55 does not.  My suspicion is that line 55 (returning the inverted yaw) is inconsistent with line 51.  We should print both out and see if they are tracking in opposite directions and decide which one is correct.  I don't think this should block this PR because it's been in the code since Veer.  Can you add a TODO to in the code to check this out?

Reply:
Ok ... will add todo

Reply:
done